packages/dids/src/resolver/universal-resolver.ts

```diff
@@ -121,16 +121,28 @@ export class UniversalResolver implements DidResolver, DidUrlDereferencer {
     }
 
     const cachedResolutionResult = await this.cache.get(parsedDid.uri);
-
     if (cachedResolutionResult) {
+      if (cachedResolutionResult?.isStaleInCache === true) {
+        // if the cache is stale, we would like to attempt refreshing it after returning the existing stale result
+        isomorphicNextTick(() => this.resolveAndCache(parsedDid.uri, resolver, options));
+      }
+
       return cachedResolutionResult;
```

Comment:
If I am reading this correctly, we alway return the stale data when `isStaleInCache` is `true`. This seems to be to aggressive, imagine there is a DID that is stale in cache for 1 year (extremely example to demonstrate the point) because it is just not used very often. A new resolution resolution will return this 1 year old DID state even though there may have been multiple updates to this DID.

Consider an alternate implementation where there is a timeout for resolving DID (say 5 seconds) when it is stale, use the new state if it is returned, and if nothing/error comes back, it refreshes the TTL using the stale data.

Reply:
Good idea, i'll implement something a little more sane for that.